src/test/java/org/jvnet/jenkins/plugins/nodelabelparameter/parameterizedtrigger/AllNodesBuildParameterFactoryTest.java

```diff
@@ -0,0 +1,84 @@
+package org.jvnet.jenkins.plugins.nodelabelparameter.parameterizedtrigger;
+
+import static org.junit.Assert.*;
+
+import hudson.model.Computer;
+import hudson.plugins.parameterizedtrigger.AbstractBuildParameters;
+import hudson.slaves.DumbSlave;
+import java.util.List;
+import java.util.Objects;
+import jenkins.model.Jenkins;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class AllNodesBuildParameterFactoryTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private AllNodesBuildParameterFactory factory;
+
+    @Before
+    public void setUp() throws Exception {
+        factory = new AllNodesBuildParameterFactory();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        Computer[] computers = Jenkins.get().getComputers();
+        for (Computer computer : computers) {
+            computer.setTemporarilyOffline(false, null);
+        }
+    }
+
+    @Test
+    public void getParameters_withAllNodesOnline() throws Exception {
+        DumbSlave node1 = j.createOnlineSlave();
+        DumbSlave node2 = j.createOnlineSlave();
+
+        List<AbstractBuildParameters> params = factory.getParameters(null, j.createTaskListener());
+
+        // trigger build on Jenkins (built-in) included in params.size() so added 1 in expected
```

Comment:
Optional phrasing change on the comment.  When I read the phrase "trigger build on Jenkins", I assume that a build will be run in the later code.  That's not what happens in this test.  This test is checking that the parameters are set as expected.
```suggestion
        // Parameters should include 2 agents and the Jenkins controller (built-in)
```